modules/gapi/include/opencv2/gapi/infer/ie.hpp

```diff
@@ -67,6 +67,17 @@ namespace detail {
         Kind kind;
         bool is_generic;
         IEConfig config;
+
+        ParamDesc(std::string model, std::string weights, std::string device,
+                  std::vector<std::string> inputNames, std::vector<std::string> outputNames,
+                  std::unordered_map<std::string, ConstInput> constInputs,
+                  std::size_t numIn, std::size_t numOut, Kind kind_, bool isGen, IEConfig config_)
```

Comment:
Use `const reference` as input parameters

Reply:
Of course. Thanks!
Fixed

Reply:
rolled back and added `std::move` in initializations to make less copies if rvalues are passed

Reply:
@OrestChura Nope, that won't work. All of those objects will be copied to the function's arguments. You should you `const reference` or universal reference (aka `&&` + `std::forward`)

Reply:
@TolyaTalamanov can you take a look?

Reply:
+ No needed implement both overloads  (const& and &&)
- In case copy, `ParamsDesc(const& T)` make only one copy but `ParamDesc(T t)` make copy + move

I'm ok with both of them

Reply:
I'd prefer to leave it as is now as it has some advantages and, actually, no so important because it's not the public code

Reply:
> because it's not the public code

Actually, it is a public header:

> modules/gapi/include/opencv2/gapi/infer/ie.hpp 

---

"copy + move" vs "const reference + copy"

The second one is preferable.
Copying in the first case is done on a caller side. Such copying significantly increases binary code size in cases of multiple use (default).
Copying after "const reference" is limited to 1 instance of binary code.

Reply:
I see! OK then, thanks. I've returned the const references already.